src/components/views/BlackCardSelection.js

```diff
@@ -72,22 +114,41 @@ const BlackCardSelection = () => {
                 setCards(response.data)
             }
             catch (error) {
-                // todo if user already has selected a Card, server should probably throw an error
                 console.error("Details:", error);
                 alert("Invalid Input:\n " + handleError(error));
             }
         }
+        fetchBlackCard();
         fetchCards();
+
     }, []);
 
+    // placeholder in case of failure
+    let textContent;
+    let cardContent;
+    let now = new Date();
+    const diffTime = Math.abs(Date.parse(now) - Date.parse(startTime));
```

Comment:
Why do you use absolute value here? creationDate cannot be in the future at any point ;). Furthermore, you don't have to parse it. Just use `const diffTime = new Date() - new Date(creationDate)` , as in Matches.js .

Reply:
thanks! done